src/hooks/use-has-waited-for-query.ts

```diff
@@ -0,0 +1,36 @@
+import { QueryKey, useQueryClient } from '@tanstack/react-query'
+
+/**
+ * Hook to return whether we've attempted a particular query yet.
+ *
+ * This is intended for use in individual progress queries,
+ * in order to check whether a relevant batch query has been attempted.
+ *
+ * Why? We want to give the batch query a chance to prime the cache
+ * before firing off individual queries. So, we likely want to disable
+ * individual queries until a batch query has been attempted at least once.
+ */
+export function useHasWaitedForQuery(queryKey: QueryKey): boolean {
```

Comment:
This allows to instruct individual tutorial and collection progress queries to wait on a batch progress query (if one has been initialized).

- If there is not a matching batch progress query according to `queryClient.getQueryState`, individual queries will not wait (and we'll see many individual progress requests to the Learn API).
- If there is a matching batch progress query, individual queries will wait till it settles in some way (and we should see only a single batch progress request to the Learn API for progress).